### PR TITLE
Feature/56 consultations page v2

### LINKED ISSUE #56: Consultation list page v2
With filters (major, subject, date, etc...)
Calendar view

### PR DESCRIPTION
**Pull request by DannySS123**
Kids, don't play with fullcalendar CSS. Not fun.

- closes #56

### CODE REVIEW

**Review by kisbogdan-kolos — APPROVED** [review]

**Review by Tschonti — COMMENTED** [review]
Very nice!

One bug I found, steps to reproduce:

1. Click on an event from the calendar
2. Go back to the previous page (with the browser's back button)
3. Click on calendar view
4. Calendar is invisible :(

Kinda weird that 6 weeks show up on the month view, even though a month is never longer than 5 weeks. It makes the calendar a bit too tall. From what I could find, it might be hard to fix this, so not a priority.

Button colors are a bit misleading, I know, they're the same as everywhere else, we should think those over, but in a separate pr.

File: src/pages/consultations/components/ConsultationsCalendarPanel.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  end: string
+  color: string
+}
+
+const StyleWrapper = styled.div`
+  --fc-button-active-bg-color: ${(props) => customTheme.colors.brand[props.className === 'light' ? 400 : 300]};
+  --fc-button-bg-color: ${(props) => customTheme.colors.brand[props.className === 'light' ? 500 : 200]};
+  --fc-button-border-color: ${(props) => customTheme.colors.brand[props.className === 'light' ? 500 : 200]};
+  --fc-button-hover-bg-color: ${(props) => customTheme.colors.brand[props.className === 'light' ? 400 : 300]};
+  --fc-button-hover-border-color: ${(props) => customTheme.colors.brand[props.className === 'light' ? 400 : 300]};
```
Comment (berenteb) [inline src/pages/consultations/components/ConsultationsCalendarPanel.tsx:31]: részvétem

File: src/pages/consultations/components/ConsultationsCalendarPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            }
+          }}
+          locale="hu"
+          firstDay={1}
+          allDaySlot={false}
+          dayMaxEvents={true}
+          weekends={true}
+          nowIndicator={true}
+          events={events}
+          buttonHints={{ week: 'Hét nézet', month: 'Hónap nézet', today: 'Ugrás a mai naphoz', next: 'Következő', prev: 'Előző' }}
```
Comment (berenteb) [inline src/pages/consultations/components/ConsultationsCalendarPanel.tsx:107]: Heti nézet*

**Review by berenteb — APPROVED** [review]

**Comment by DannySS123** [comment]
@Tschonti  I tried many times, but could not reproduce the bug. It works well for me.

fixedWeekCount solved your issue with having too many weeks in a month view. But even if there are no months with 6 weeks, becaouse of the fixed view there will be some months that show 6 weeks. For example:
image
But all the others that fit will show only 5 weeks:
image
Hope it's a bit better this way.

**Review by Tschonti — APPROVED** [review]
Much better, we'll investigate the bug later

**Comment by DannySS123** [comment]
Fun fact: about every 10 years there might be a 4 week month 💀 
image
image

**Pull request merged**